dhcpv4/nclient4/client.go

```diff
@@ -233,10 +234,16 @@ func withBufferCap(n int) ClientOpt {
 
 // WithRetry configures the number of retransmissions to attempt.
 //
+// 0 retries means infinite retries.
+//
 // Default is 3.
 func WithRetry(r int) ClientOpt {
```

Comment:
what about requiring an uint here? The user will get a compile time error if a negative integer is passed, a bit annoying (more typing) but safer

Reply:
I didn't want to disturb existing users, haha